tests/pfc/test_pfc_pause_lossy.py

```diff
@@ -0,0 +1,106 @@
+
+START_DELAY = [1]
+TRAFFIC_DURATION = [3]
+PAUSE_LINE_RATE = [100]
+TRAFFIC_LINE_RATE = [50]
+BW_MULTIPLIER = [1000000]
+FRAME_SIZE = [1024]
+TOLERANCE_THRESHOLD = .97
+
+@pytest.mark.parametrize('start_delay', START_DELAY)
+@pytest.mark.parametrize('traffic_duration', TRAFFIC_DURATION)
+@pytest.mark.parametrize('pause_line_rate', PAUSE_LINE_RATE)
+@pytest.mark.parametrize('traffic_line_rate', TRAFFIC_LINE_RATE)
+@pytest.mark.parametrize('bw_multiplier', BW_MULTIPLIER)
+@pytest.mark.parametrize('frame_size', FRAME_SIZE)
+
+def test_pfc_pause_lossy_traffic(api, 
+                                 duthost, 
+                                 lossy_configs, 
+                                 start_delay,
+                                 pause_line_rate,
+                                 traffic_line_rate, 
+                                 traffic_duration,
+                                 port_bandwidth,
+                                 frame_size) :
+    """
+    This test case checks the behaviour of the SONiC DUT when it receives 
+    a PFC pause frame on lossy priorities.
+
+                                +-----------+
+    [Keysight Chassis Tx Port]  |           | [Keysight Chassis Rx Port]
+    --------------------------->| SONiC DUT |<---------------------------
+    Test Data Traffic +         |           |  PFC pause frame on 
+    Background Dada Traffic     +-----------+  "lossy" priorities.
+
+    1. Configure SONiC DUT with multipul lossless priorities. 
+    2. On SONiC DUT enable PFC on several lossless priorities e.g priority 
+       3 and 4.
+    3. On the Keysight chassis Tx port create two flows - a) 'Test Data Traffic'
+       and b) 'Background Data traffic'.
+    4. Configure 'Test Data Traffic' such that it contains traffic items
+       with all lossy priorities.
+    5. Configure 'Background Data Traffic' it contains traffic items with
+       all lossless priorities.
+    6. From Rx port send pause frames on all lossless priorities. Then
+       start 'Test Data Traffic' and 'Background Data Traffic'.
+    7. Verify the following: 
+       (a) When Pause Storm are running, Keysight Rx port is receiving
+       both 'Test Data Traffic' and 'Background Data traffic'.
+       (b) When Pause Storm are stoped, then also Keysight Rx port is receiving
+       both 'Test Data Traffic' and 'Background Data traffic'.
+    """
+    duthost.shell('sudo pfcwd stop')
+
+    for base_config in lossy_configs:
+        # create the configuration
+        api.set_state(State(ConfigState(config=base_config, state='set')))
+
+        # start all flows
+        api.set_state(State(FlowTransmitState(state='start')))
+
+        exp_dur = start_delay + traffic_duration
+        logger.info("Traffic is running for %s seconds" %(exp_dur))
+        time.sleep(exp_dur)
+
+        # stop all flows
+        api.set_state(State(FlowTransmitState(state='stop')))
+
+        # Get statistics
+        stat_captions =['Test Data', 'Background Data']
```

Comment:
we need to define 'Test Data' and 'Background Data' in tests/pfc/files/configs/pfc.py

Reply:
I have defined 'Test Data' and 'Background Data' in pfc/test_pfc_pause_lossy.py and made them available in pfc.py as parameter